pages/api/events/class/index.ts

```diff
@@ -0,0 +1,91 @@
+import { NextApiHandler, NextApiRequest, NextApiResponse } from "next";
+import { createClassEvent } from "../../../../lib/database/events";
+import { createCalenderEvent } from "../../../../lib/database/calender";
+import { createCommitment } from "../../../../lib/database/commitments";
+import { CreateClassEvent, ClassEvent, CreateClassEventSchema } from "../../../../models/events";
+import { decode } from "io-ts-promise";
+import { StatusCodes } from "http-status-codes";
+import { rrulestr } from "rrule";
+import { DateTime } from "luxon";
+
+// handles requests to /api/events/class
+const eventHandler: NextApiHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+  switch (req.method) {
+    case "POST": {
+      let newEvent: CreateClassEvent;
+      try {
+        newEvent = await decode(CreateClassEventSchema, req.body);
+      } catch (e) {
+        return res.status(StatusCodes.BAD_REQUEST).json("Fields are not correctly entered");
+      }
+      try {
+        const result = await createClassEvent(
+          newEvent.name,
+          newEvent.startTime,
+          newEvent.endTime,
+          newEvent.timeZone,
+          newEvent.rrule,
+          newEvent.language,
+          newEvent.neverEnding,
+          newEvent.backgroundColor,
+          newEvent.teachers
+        );
+
+        const ruleObj = rrulestr(newEvent.rrule);
+        const allDates = newEvent.neverEnding ? ruleObj.all().splice(0, 365) : ruleObj.all();
+        const startTime = DateTime.fromISO(newEvent.startTime);
+        const endTime = DateTime.fromISO(newEvent.endTime);
+
+        // Loops through all dates and inserts into calender_information table
+        for (const date of allDates) {
+          const dateStart = DateTime.fromJSDate(date, { zone: newEvent.timeZone }).set({
+            hour: startTime.hour,
+            minute: startTime.minute,
+            second: startTime.second,
+          });
+
+          const dateEnd = DateTime.fromJSDate(date, { zone: newEvent.timeZone }).set({
+            hour: endTime.hour,
+            minute: endTime.minute,
+            second: endTime.second,
+          });
```

Comment:
Sadly, the `rrule` is dumb and auto inserts time when your not supposed to. It seems like that is interfering with the timezone being set. So, we just have to construct a `Date` object without the time. Something like this should do the trick:
``` javascript
const dateWithoutTime = new Date(date.getFullYear(), date.getMonth(), date.getDate());
```
Now, instead of using `date` to the `DateTime` object, use `dateWithoutTime`